pkg/controller/nodeipam/node_ipam_controller_test.go

```diff
@@ -92,18 +92,21 @@ func TestNewNodeIpamControllerWithCIDRMasks(t *testing.T) {
 		{"invalid_serviceCIDR_contains_clusterCIDR", "10.0.0.0/23", "10.0.0.0/21", emptyServiceCIDR, []int{24}, ipam.IPAMFromClusterAllocatorType, true},
 		{"invalid_CIDR_mask_size", "10.0.0.0/24,2000::/64", "10.1.0.0/21", emptyServiceCIDR, []int{24, 48}, ipam.IPAMFromClusterAllocatorType, true},
 	} {
-		t.Run(tc.desc, func(t *testing.T) {
-			clusterCidrs, _ := netutils.ParseCIDRs(strings.Split(tc.clusterCIDR, ","))
-			_, serviceCIDRIpNet, _ := netutils.ParseCIDRSloppy(tc.serviceCIDR)
-			_, secondaryServiceCIDRIpNet, _ := netutils.ParseCIDRSloppy(tc.secondaryServiceCIDR)
+		test := tc
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			clusterCidrs, _ := netutils.ParseCIDRs(strings.Split(test.clusterCIDR, ","))
+			_, serviceCIDRIpNet, _ := netutils.ParseCIDRSloppy(test.serviceCIDR)
+			_, secondaryServiceCIDRIpNet, _ := netutils.ParseCIDRSloppy(test.secondaryServiceCIDR)
 
 			if os.Getenv("EXIT_ON_FATAL") == "1" {
 				// This is the subprocess which runs the actual code.
-				newTestNodeIpamController(clusterCidrs, serviceCIDRIpNet, secondaryServiceCIDRIpNet, tc.maskSize, tc.allocatorType)
+				newTestNodeIpamController(clusterCidrs, serviceCIDRIpNet, secondaryServiceCIDRIpNet, test.maskSize, test.allocatorType)
 				return
 			}
 			// This is the host process that monitors the exit code of the subprocess.
-			cmd := exec.Command(os.Args[0], "-test.run=TestNewNodeIpamControllerWithCIDRMasks/"+tc.desc)
+			cmd := exec.Command(os.Args[0], "-test.run=TestNewNodeIpamControllerWithCIDRMasks/"+test.desc)
```

Comment:
why this test executes itself?

Reply:
I see, the controller panics

```
			klog.Fatal("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
```

Reply:
#68952 added this.
It seems to be not that graceful.

Reply:
the test is wrong, let me check it better

Reply:
some of the test cases are wrong, checking right now